core/runtime/v2/bundle.go

```diff
@@ -131,7 +131,7 @@ func (b *Bundle) Delete() error {
 	if err := mount.UnmountRecursive(rootfs, 0); err != nil {
 		return fmt.Errorf("unmount rootfs %s: %w", rootfs, err)
 	}
-	if err := os.Remove(rootfs); err != nil && !os.IsNotExist(err) {
```

Comment:
Could you please share the mount points on that rootfs? It's unlikely for a directory to be non-empty after a successful unmount. Without reproduction steps, I won’t vote for it since the change is unclear and could unexpectedly remove content from the rootfs.